src/entities/TransactionEntity.ts

```diff
@@ -0,0 +1,60 @@
+import { Column, CreateDateColumn, Entity, PrimaryColumn, UpdateDateColumn } from 'typeorm';
+
+import { ONE_SECOND_MS } from '../constants';
+
+import { TransactionEntityOpts } from './types';
+
+@Entity({ name: 'transactions' })
+export class TransactionEntity {
+    @PrimaryColumn({ name: 'ethereum_tx_hash', type: 'varchar' })
+    public hash: string;
+
+    @Column({ name: 'status', type: 'varchar' })
+    public status: string;
+
+    @Column({ name: 'expected_mined_in_sec', type: 'int' })
+    public expectedMinedInSec?: number;
+
+    @Column({ name: 'nonce', type: 'varchar' })
+    public nonce: string;
+
+    @Column({ name: 'gas_price', type: 'varchar' })
+    public gasPrice: string;
+
+    @Column({ name: 'meta_txn_relayer_address', type: 'varchar' })
+    public metaTxnRelayerAddress: string;
+
+    @CreateDateColumn({ name: 'created_at' })
+    // tslint:ignore-next-line
+    public createdAt?: Date;
+
+    @UpdateDateColumn({ name: 'updated_at' })
+    public updatedAt?: Date;
+
+    @Column({ name: 'expected_at', type: 'timestamptz' })
+    public expectedAt: Date;
+
+    // HACK(oskar) we want all fields to be set, otherwise we should not accept
+    // a transaction entity, however because of this issue:
+    // https://github.com/typeorm/typeorm/issues/1772 we cannot accept undefined
+    // as an argument to the constructor, to not break migrations with serialize
```

Comment:
If we throw inside of the constructor `typeorm` will fail on auto-migration (the `serialize` option does this migration).

We could have a static method as suggested in the issue that would throw an error, however I don't know how viable that is.

Reply:
Hm, so how do we ensure these entries cannot be stored with empty-string values?

Reply:
Well, the best way is to get rid of `synchronize: true` and have a proper migration setup, but that probably is outside of the scope of this PR.

An alternative is to make the `constructor` private and have a `public static` method `create` or `make` which would throw errors on input validation, so instead of:
```
new TransactionEntity({..})
```

it would be:
```
TransacitonEntity.create({})
```

Thoughts?

Reply:
I like that as a short-term solution. Let's do that for now.

Reply:
added `public static make`